### PR TITLE
Fix Memory Error when data is larger than chunk size in paint().

### PR DESCRIPTION
**Pull request by rainwoodman**
reported by @zdplayground

See
https://github.com/bccp/nbodykit/commit/4e41aab09c4c7caa702fe7b9a31f1b54c5923f07#commitcomment-25540401

### CODE REVIEW

**Comment by rainwoodman** [comment]
@zdplayground : 

You can test this after running:
```
 pip install -U --no-deps https://github.com/rainwoodman/nbodykit/archive/large-data.zip
```
In your conda environment of nbodykit; then restart the kernel.

**Comment by zdplayground** [comment]
@rainwoodman:
After installing large-data.zip, the memory error disappears, though the running time of FFTPower for such case seems a little bit long.

File: nbodykit/base/catalogmesh.py [hunk 1]
```diff
@@ -340,30 +341,26 @@ def to_real_field(self, out=None, normalize=True):
...
 
             if len(Position) != 0:
 
+                # selection has to be computed many times when data is `large`.
+                sel = self.base.compute(Selection[s])
+
                 # be sure to use the source to compute
                 position, weight, value = \
-                    self.base.compute(Position[s], Weight[s], Value[s])
+                    self.base.compute(Position[s][sel], Weight[s][sel], Value[s][sel])
```
Comment (nickhand) [inline nbodykit/base/catalogmesh.py]: I think we might want to apply `` sel`` to the arrays after computing to be safer (this was the logic used in 4e41aab0).

**Comment by rainwoodman** [comment]
@zdplayground  it's 1 billion particles on a single core; that's why I usually run this from the job queue with ~ 100 cores.

**Comment by rainwoodman** [comment]
@nickhand let's wait till travis is green before merging.

**Pull request merged**